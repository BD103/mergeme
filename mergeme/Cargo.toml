[package]
name = "mergeme"
version = "0.1.0"
edition = "2024"

[features]
default = ["derive"]
derive = ["dep:mergeme_derive"]

[dependencies]
mergeme_derive = { path = "../mergeme_derive", version = "0.1.0", optional = true }
