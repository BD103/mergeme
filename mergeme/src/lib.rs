#![no_std]

#[cfg(feature = "derive")]
pub use mergeme_derive::Merge;

pub trait Merge<Partial>: Sized {
    fn merge_in_place(&mut self, other: Partial);

    fn merge(mut self, other: Partial) -> Self {
        self.merge_in_place(other);
        self
    }
}
