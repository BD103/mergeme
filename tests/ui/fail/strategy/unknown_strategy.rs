use mergeme::Merge;

#[derive(Merge)]
#[partial(PartialDog)]
struct Dog {
    name: String,
    // `add` is not a valid strategy.
    #[strategy(add)]
    age: u16,
}

fn main() {}
