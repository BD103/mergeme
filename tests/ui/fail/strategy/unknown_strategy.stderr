error: expected `#[strategy(overwrite)]` or `#[strategy(merge)]`
 --> tests/ui/fail/strategy/unknown_strategy.rs:8:16
  |
8 |     #[strategy(add)]
  |                ^^^
